accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/helper/runtime/RuntimeHelper.java

```diff
@@ -0,0 +1,76 @@
+package biz.netcentric.cq.tools.actool.helper.runtime;
+
+import java.util.Set;
+
+import javax.jcr.Node;
+import javax.jcr.NodeIterator;
+import javax.jcr.Property;
+import javax.jcr.PropertyIterator;
+import javax.jcr.RepositoryException;
+import javax.jcr.Session;
+
+import org.apache.sling.settings.SlingSettingsService;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.Constants;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.startlevel.FrameworkStartLevel;
+
+/** Provides basic context information about the runtime. In own package so it can be inlined with "Conditional-Package" from startup bundle */
+public class RuntimeHelper {
+    
+    private static final String RUNMODE_CLOUD_READY = "cloud-ready";
+
+    public static boolean isCompositeNodeStore(Session session) {
+        
+        try {
+            Node appsNode = session.getNode("/apps");
+            // see https://issues.apache.org/jira/browse/OAK-6563
+            boolean isCompositeNode = !session.hasCapability("addNode", appsNode, new Object[] { "nt:folder" });
```

Comment:
This could also mean that the given session just does not have the according rights, isn't it? That should be explicitly mentioned or a check for the according rights must be added here as well. 
For simplicity reasons I would always rely on the service user internally and not pass the session as argument....

Reply:
bq. For simplicity reasons I would always rely on the service user internally and not pass the session as argument....

The session can be either from the install hook (packmgr) or the service session  - I really want to make sure that the exact session that is also used for installation is used here

bq. This could also mean that the given session just does not have the according rights, isn't it? 
Correct, will try a combination with `session.hasPermission()` to ensure it's 100% correct

Reply:
Improved implementation: https://github.com/Netcentric/accesscontroltool/blob/6db01125d0c038cacc2df27b908ad709b129e026/accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/helper/runtime/RuntimeHelper.java#L41